### PR TITLE
refactor: generally move things around to be easier

### PR DESCRIPTION
**Pull request by intellix**
During my quest to debug why Falcon environments don't always get destroyed...

Split the 2 different modules into 2 folders and moved a few bits around to try and simplify the flow. Still a work in progress but should be mergeable if you agree or have any ideas

### CODE REVIEW

File: modules/external/controller/Falcon.ts [hunk 1]
```diff
@@ -115,29 +115,6 @@ class Falcon {
...
-          git.removeLabel(prNumber, fail);
-        }
-
-        if (item.name === pending && toBeAdded !== pending) {
-          git.removeLabel(prNumber, pending);
-        }
-      });
-      git.addLabels(prNumber, [toBeAdded]);
-    });
-  }
```
Comment (intellix) [inline modules/external/controller/Falcon.ts:140]: Was trying to simplify the flow and saw this being called. I didn't see what it had to do with Falcon management :P Removes it's dependency on git

File: modules/external/routes/pullrequest.ts [hunk 2]
```diff
@@ -46,18 +45,13 @@ router.post('/', (req, res) => {
...
-            'falcon:destroy',
-            falcon.destroy(pr, project),
-          ),
-        );
+      if (pr.isMerged()) {
+        git.getIssues(pr, (issues: string[]) => bot.handleMerged(issues));
+      }
+
+      bot.purgeScreenshots(pr.branch);
+      bot.destroyAll(pr);
```
Comment (intellix) [inline modules/external/routes/pullrequest.ts:53]: Noticed this was the 2nd place where websockets are being manipulated directly so wanted to centralise them into the Bot and make that websocket private

File: modules/external/index.ts [hunk 3]
```diff
@@ -12,17 +23,14 @@ app.use(errorHandler());
...
     app.get("env")
   );
   console.log('  Press CTRL-C to stop\n');
 });
 
-
-
 bot.setWebsocket(socket(server));
 
-export default server;
```
Comment (intellix) [inline modules/external/index.ts:28]: I renamed this file to index.ts as I think it makes it easier to understand the flow. It's the entrypoint and common practise for it to be the index

**Comment by intellix** [comment]
Some thoughts on naming. I tried to keep them the opposite of each other:
- internal -> external
- public -> private
- inside -> outside

**Review by picheli20 — APPROVED** [review]

**Pull request merged**